source/developer_manual/design/dict_protocol.rst

```diff
@@ -0,0 +1,251 @@
+.. _dovecot_dict_protocol:
+
+=====================
+Dovecot Dict Protocol
+=====================
+
+General
+-------
+
+Dovecot's dict protocol is a line based protocol between the dict process
+acting as the **server** and the dict proxy process acting as the **client**.
+The dict server receives lines containing commands and responds to the client
+with results. Each line ends with an LF character and the maximum line the
+client is allowed to send a command is 65536 (64KB of data).
+Maximum line length for dict server's response line is 131072 (128KB of data).
+
+Each command is in format:
+
+.. code-block:: none
+
+   <command short name><parameters separated with TAB>
+
+The command short name is a single character and the rest of the line
+containing the command parameters are followed immediately without any
+whitespace in between. 
+
+The commands available in dict protocol and their corresponding short name
+is shown in table bellow.
+
++--------------------+---------------------+
+| Command            | Command short name  |
++====================+=====================+
+| HELLO              |  H                  |
++--------------------+---------------------+
+| LOOKUP             |  L                  |
++--------------------+---------------------+
+| ITERATE            |  I                  |
++--------------------+---------------------+
+| BEGIN              |  B                  |
++--------------------+---------------------+
+| COMMIT             |  C                  |
++--------------------+---------------------+
+| COMMIT_ASYNC       |  D                  |
++--------------------+---------------------+
+| ROLLBACK           |  R                  |
++--------------------+---------------------+
+| SET                |  S                  |
++--------------------+---------------------+
+| UNSET              |  U                  |
++--------------------+---------------------+
+| ATOMIC_INC         |  A                  |
++--------------------+---------------------+
+| TIMESTAMP          |  T                  |
++--------------------+---------------------+
+
+The reply from dict server is in the same format.
+
+.. code-block:: none
+
+   <result short status>(<optional extra values separated with TAB>)
+
+
+Result statuses and their corresponding short name is shown in table bellow.
+
++--------------------+---------------------+
+| Result status      | Status short name   |
++====================+=====================+
+| OK                 |  O                  |
++--------------------+---------------------+
+| MULTI_OK           |  M                  |
++--------------------+---------------------+
```

Comment:
It's a bit difficult to know what results are valid for which commands. Best would be to move the table to each command and list only the valid statuses. Or I'm not sure if a list would be better. Also some of these take parameters. Especially the LOOKUP command's OK/MULTI_OK parameters need explaining. And ITERATE's output.

Reply:
Added. Though, I think async responses aren't necessary since they are not used.